src/js/player.js

```diff
@@ -1843,6 +1872,31 @@ class Player extends Component {
       return this.techGet_('src');
     }
 
+    let src = source;
+
+    if (Array.isArray(source)) {
+      this.cache_.sources = source;
+      src = source[0];
+    } else if (typeof source === 'string') {
+      src = {
+        src: source
+      };
+
+      this.cache_.sources = [src];
+    }
+
+    this.cache_.source = src;
+    this.cache_.src = src.src;
+
+    middleware.setSource(src, (tech, src_, mws) => {
+      this.middleware_ = mws;
+      this.loadTech_(tech.name, src_);
+      middleware.setTech(mws, this.tech_);
+    });
```

Comment:
If `tech.name` is the only thing used here, we could destructure in the args. I don't feel strongly about this one, just something I've started doing a little more of lately.

```
middleware.setSource(src, ({ name }, src_, mws) => {
  this.middleware_ = mws;
  this.loadTech_(name, src_);
  middleware.setTech(mws, this.tech_);
});
```